src/navigation/reroute.js

```diff
@@ -69,8 +69,8 @@ export function reroute(pendingPromises = [], eventArguments) {
     return loadApps();
   }
 
-  function cancelNavigation() {
-    navigationIsCanceled = true;
+  function cancelNavigation(fn) {
+    navigationIsCanceled = typeof fn === "function" ? fn : true;
```

Comment:
This assumes that `cancelNavigation` is called only once, which does not hold true. It's quite possible for multiple single-spa applications to each add an event listener and call cancelNavigation. To solve, we'd need to store each of the functions in an array and then do a Promise.all() to wait for all of them before knowing whether the navigation is canceled.